certora/specs/AllocationMorphoMarketV1Adapter.spec

```diff
@@ -0,0 +1,159 @@
+
+    require (MorphoMarketV1Adapter.allocation(marketParams) <= MorphoMarketV1.totalSupplyAssets(marketId), "assume the adapter's allocation are less than or equal to the total assets");
+    require (MorphoMarketV1Adapter.shares(marketId) <= MorphoMarketV1.totalSupplyShares(marketId), "assume the adapter's shares are less than or equal to the total shares");
+
+    // Ensure the VaultV2 and Morpho contracts are properly linked to the adapter in the conf file.
+    assert MorphoMarketV1Adapter.parentVault == currentContract;
+    assert MorphoMarketV1Adapter.morpho == MorphoMarketV1;
+
+    allocate(e, MorphoMarketV1Adapter, data, assets);
+    realizeLoss(e, MorphoMarketV1Adapter, data);
+
+    assert MorphoMarketV1Adapter.allocation(marketParams) == Utils.expectedSupplyAssets(e, MorphoMarketV1, marketParams, MorphoMarketV1Adapter.shares(marketId));
+}
+
+rule deallocateMorphoMarketV1Adapter(env e, bytes data, uint256 assets) {
+    // Trick to require that all the following addresses are different.
+    require (MorphoMarketV1 == 0x10, "ack");
+    require (MorphoMarketV1Adapter == 0x11, "ack");
+    require (currentContract == 0x12, "ack");
+
+    require (forall bytes32 id . ghostAllocationBefore[id] == ghostAllocationAfter[id], "setup allocation before to be equal to allocation after");
+
+    MorphoHarness.MarketParams marketParams;
+    (marketParams, _) = Utils.morphoMarketV1MarketParams(data);
+
+    // Ensure the VaultV2 and Morpho contracts are properly linked to the adapter in the conf file.
+    assert MorphoMarketV1Adapter.parentVault == currentContract;
+    assert MorphoMarketV1Adapter.morpho == MorphoMarketV1;
+
+    bytes32[] adapterIds = MorphoMarketV1Adapter.ids(marketParams);
+
+    deallocate(e, MorphoMarketV1Adapter, data, assets);
+
+    assert forall uint i. i < adapterIds.length => ghostAllocationAfter[adapterIds[i]] == ghostAllocationBefore[adapterIds[i]] + ghostInterest - assets;
+    assert forall bytes32 id. (forall uint i . i >= adapterIds.length || id != adapterIds[i]) => ghostAllocationAfter[id] == ghostAllocationBefore[id];
+}
+
+rule deallocateMorphoMarketV1AdapterAllocationVsExpectedAssets(env e, bytes data, uint256 assets) {
+    // Trick to require that all the following addresses are different.
+    require (MorphoMarketV1 == 0x10, "ack");
+    require (MorphoMarketV1Adapter == 0x11, "ack");
+    require (currentContract == 0x12, "ack");
+
+    MorphoHarness.MarketParams marketParams;
+    MorphoHarness.Id marketId;
+    (marketParams, marketId) = Utils.morphoMarketV1MarketParams(data);
+
+    require (MorphoMarketV1Adapter.allocation(marketParams) <= MorphoMarketV1.totalSupplyAssets(marketId), "assume the adapter's allocation is less than or equal to the total assets");
+    require (MorphoMarketV1Adapter.shares(marketId) <= MorphoMarketV1.totalSupplyShares(marketId), "assume the adapter's shares are less than or equal to the total shares");
```

Comment:
Same, see [this run](https://prover.certora.com/output/66603/0e1877abb7a948669cdae6bfcc93569c?anonymousKey=715d01bc3569880b277de17229203a3561bdf690)